src/text/raw_text_reader.rs

```diff
@@ -1,476 +1,59 @@
-        // If the reader's current value is the beginning of a container and the user calls `next()`,
-        // we need to skip the entire container. We can do this by stepping into and then out of
-        // that container; `step_out()` has logic that will exhaust the remaining values.
-        let need_to_skip_container = !self.is_null()
-            && self
-                .current_value
-                .as_ref()
-                .map(|v| v.value().ion_type().is_container())
-                .unwrap_or(false);
-
-        if need_to_skip_container {
-            self.step_in()?;
-            self.step_out()?;
-        }
-
-        // Unset variables holding onto information about the previous position.
-        self.current_ivm = None;
-        self.current_value = None;
-        self.current_field_name = None;
-
-        if self.parents.is_empty() {
-            // The `parents` stack is empty. We're at the top level.
-
-            // If the reader has already found EOF (the end of the top level), there's no need to
-            // try to read more data. Return Ok(None).
-            if self.is_eof {
-                self.current_value = None;
-                return Ok(());
-            }
-
-            let next_stream_item = self.parse_next_nom(stream_item);
-            return self.process_stream_item(next_stream_item);
-        }
-
-        // Otherwise, the `parents` stack is not empty. We're inside a container.
-
-        // The `ParentLevel` type is only a couple of stack-allocated bytes. It's very cheap to clone.
-        let parent = *self.parents.last().unwrap();
-        // If the reader had already found the end of this container, return Ok(None).
-        if parent.is_exhausted() {
-            self.current_value = None;
-            return Ok(());
-        }
-        // Otherwise, try to read the next value. The syntax we expect will depend on the
-        // IonType of the parent container.
-        let value = match parent.ion_type() {
-            IonType::List => self.next_list_value(),
-            IonType::SExpression => self.next_s_expression_value(),
-            IonType::Struct => {
-                // If the reader finds a field name...
-                if let Some(field_name) = self.next_struct_field_name()? {
-                    // ...remember it and return the field value that follows.
-                    self.current_field_name = Some(field_name);
-                    let field_value_result = self.next_struct_field_value()?;
-                    Ok(Some(field_value_result))
-                } else {
-                    // Otherwise, this is the end of the struct.
-                    Ok(None)
-                }
-            }
-            other => unreachable!(
-                "The reader's `parents` stack contained a scalar value: {:?}",
-                other
-            ),
+    pub fn new_with_size(input: T, size: usize) -> IonResult<RawTextReader<T>> {
```

Comment:
Can you open an issue to track exposing this to the `Builder` API?

Reply:
Absolutely